packages/ckeditor5-bbcode/src/rules/BBCodeUrl.ts

```diff
@@ -0,0 +1,35 @@
+import { BBCodeProcessingRule } from "./BBCodeProcessingRule";
+import { isHTMLAnchorElement } from "@coremedia/ckeditor5-dom-support";
+
+/**
+ * Escapes the given URL, so that it does not contain open or close bracket.
+ *
+ * @param href - HREF attribute to encode for use as BBCode attribute value
+ */
+const escapeHref = (href: string): string => href.replace(/\[/g, "%5B").replace(/]/g, "%5D");
+
+export class BBCodeUrl implements BBCodeProcessingRule {
+  readonly id = "url";
+  readonly tags = ["url"];
+
+  toData(element: HTMLElement, content: string): undefined | string {
+    if (!isHTMLAnchorElement(element)) {
+      return;
+    }
+    // Need to use `getAttribute` instead of HTMLAnchorElement.href as the
+    // latter one may contain an already parsed `href`, that, for example,
+    // already resolved relative links to absolute links.
+    const href = element.getAttribute("href");
+    if (href) {
```

Comment:
Exit early and/or return `undefined` explicit. Currently undefined is returned implicitly.

Reply:
Ok, accepted, but not "do it sometimes this way, and sometimes the other way". Enabled:

* [consistent-return - ESLint - Pluggable JavaScript Linter](https://eslint.org/docs/latest/rules/consistent-return)

And fixed all related issues subsequently.